internal/services/recoveryservices/recovery_services_vault_guard_proxy_resource.go

```diff
@@ -0,0 +1,180 @@
+				}
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return tf.ImportAsExistsError("azurerm_recovery_services_vault_guard_proxy", id.ID())
+			}
+
+			proxy := resourceguardproxy.ResourceGuardProxyBaseResource{
+				Id:   pointer.To(id.ID()),
+				Type: pointer.To(VaultGuardResourceType),
+				Properties: pointer.To(resourceguardproxy.ResourceGuardProxyBase{
+					ResourceGuardResourceId: pointer.To(plan.ResourceGuardId),
+				}),
+			}
+
+			_, err = client.Put(ctx, id, proxy)
+			if err != nil {
+				return fmt.Errorf("creating %s:%w", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r VaultGuardProxyResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.RecoveryServices.ResourceGuardProxyClient
+
+			id, err := resourceguardproxy.ParseBackupResourceGuardProxyID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %q:%+v", metadata.ResourceData.Id(), err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("creating %s:%+v", id, err)
+			}
+
+			vaultId := vaults.NewVaultID(id.SubscriptionId, id.ResourceGroupName, id.VaultName)
+			state := VaultGuardProxyModel{
+				VaultId: vaultId.ID(),
+				Name:    id.BackupResourceGuardProxyName,
+			}
+
+			if resp.Model != nil && resp.Model.Properties != nil {
+				state.ResourceGuardId = pointer.From(resp.Model.Properties.ResourceGuardResourceId)
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r VaultGuardProxyResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var plan VaultGuardProxyModel
+			if err := metadata.Decode(&plan); err != nil {
+				return fmt.Errorf("decoding %w", err)
+			}
+			client := metadata.Client.RecoveryServices.ResourceGuardProxyClient
+
+			id, err := resourceguardproxy.ParseBackupResourceGuardProxyID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %q:%+v", metadata.ResourceData.Id(), err)
+			}
+
+			guardId, err := resourceguards.ParseResourceGuardID(plan.ResourceGuardId)
+			if err != nil {
+				return fmt.Errorf("parsing %q:%+v", plan.ResourceGuardId, err)
+			}
+
+			requestId := resourceguards.NewDeleteResourceGuardProxyRequestID(guardId.SubscriptionId, guardId.ResourceGroupName, guardId.ResourceGuardName, VaultGuardProxyDeleteRequestName)
+			_, err = client.UnlockDelete(ctx, *id, resourceguardproxy.UnlockDeleteRequest{
+				ResourceGuardOperationRequests: pointer.To([]string{requestId.ID()}),
+			})
+			if err != nil {
+				return fmt.Errorf("unlocking delete %s:%+v", id, err)
+			}
+
+			_, err = client.Delete(ctx, *id)
+			if err != nil {
```

Comment:
as it is "NewDeleteResourceGuardProxyRequestID"

should we be polling here?

these two lines can also be combined